test/SadTest.cc

```diff
@@ -802,4 +847,671 @@ INSTANTIATE_TEST_CASE_P(LOOPSAD, SadLoopTest,
+  public:
+    ExtSadCalculationTest()
+        : SADTestBase(16, 16, TEST_GET_PARAM(0), TEST_GET_PARAM(1)) {
+        src_stride_ = ref1_stride_ = 256;
+    };
+
+  protected:
+    void check_get_Extsad8x8() {
+        uint32_t best_sad8x8_1[4] = {
+            BEST_SAD_MAX, BEST_SAD_MAX, BEST_SAD_MAX, BEST_SAD_MAX};
+        uint32_t best_mv8x8_1[4] = {0};
+        uint32_t best_sad16x16_1 = UINT_MAX, best_mv16x16_1 = 0;
+        uint32_t sad16x16_1 = UINT_MAX;
+        uint32_t sad8x8_1[4] = {UINT_MAX, UINT_MAX, UINT_MAX, UINT_MAX};
+
+        prepare_data();
+
+        ext_sad_calculation_8x8_16x16(src_aligned_,
+                                      src_stride_,
+                                      ref1_aligned_,
+                                      ref1_stride_,
+                                      best_sad8x8_1,
+                                      &best_sad16x16_1,
+                                      best_mv8x8_1,
+                                      &best_mv16x16_1,
+                                      0,
+                                      &sad16x16_1,
+                                      sad8x8_1,
+                                      false);
+
+        uint32_t best_sad8x8_2[4] = {
+            BEST_SAD_MAX, BEST_SAD_MAX, BEST_SAD_MAX, BEST_SAD_MAX};
+        uint32_t best_mv8x8_2[4] = {0};
+        uint32_t best_sad16x16_2 = UINT_MAX, best_mv16x16_2 = 0;
+        uint32_t sad16x16_2 = UINT_MAX;
+        uint32_t sad8x8_2[4] = {UINT_MAX, UINT_MAX, UINT_MAX, UINT_MAX};
+
+        ext_sad_calculation_8x8_16x16_avx2_intrin(src_aligned_,
+                                                  src_stride_,
+                                                  ref1_aligned_,
+                                                  ref1_stride_,
+                                                  best_sad8x8_2,
+                                                  &best_sad16x16_2,
+                                                  best_mv8x8_2,
+                                                  &best_mv16x16_2,
+                                                  0,
+                                                  &sad16x16_2,
+                                                  sad8x8_2,
+                                                  false);
+
+        EXPECT_EQ(0,
+                  memcmp(best_sad8x8_1, best_sad8x8_2, sizeof(best_sad8x8_1)))
+            << "compare best_sad8x8 error";
+        EXPECT_EQ(0, memcmp(best_mv8x8_1, best_mv8x8_2, sizeof(best_mv8x8_1)))
+            << "compare best_mv8x8 error";
+        EXPECT_EQ(best_sad16x16_1, best_sad16x16_2)
+            << "compare best_sad16x16 error";
+        EXPECT_EQ(best_mv16x16_1, best_mv16x16_2)
+            << "compare best_mv16x16 error";
+        EXPECT_EQ(0, memcmp(sad8x8_1, sad8x8_2, sizeof(sad8x8_1)))
+            << "compare sad8x8 error";
+        EXPECT_EQ(sad16x16_1, sad16x16_2) << "compare sad16x16 error";
+    }
+
+    void check_get_Extsad32x32() {
```

Comment:
ExtSad?

Reply:
I took the first three letter of the fun I test to named this, should I change another one?

Reply:
Ok